wavelength-plugins/roomshop.js

```diff
@@ -0,0 +1,207 @@
+			if (!this.can("roommod", null, room)) return false;
+			if (!target) return this.parse(`/help roomshop`);
+			let roomshop = Db.roomshop.get(room.id, {items: {}});
+			if (!roomshop.items[toId(target)]) return this.errorReply(`The item "${target}" does not exist.`);
+			delete roomshop.items[toId(target)];
+			Db.roomshop.set(roomshop);
+			room.add(`${target} was removed from the roomshop.`);
+			this.privateModAction(`${user.name} has removed "${target}" from the roomshop.`);
+		},
+
+		"": "display",
+		shop: "display",
+		display: function (target, room, user) {
+			if (!this.runBroadcast()) return;
+			let roomshop = Db.roomshop.get(room.id, {items: {}});
+			if (!Db.roomshop.has(room.id)) return this.errorReply(`${room.title} does not have a roomshop.`);
+			let display = `<div style="max-height: 200px; width: 100%; overflow: scroll;"><table><tr><center><h1>${room.title}'s Shop!</h1></center></tr>`;
+			for (let i in roomshop.items) {
+				display += `<tr><td style="border: 2px solid #000000; width: 20%; text-align: center"><button class="button" name="send" value="/roomshop buy ${roomshop.items[i].name}">Buy Item: ${roomshop.items[i].name}</button></td><td style="border: 2px solid #000000; width: 20%; text-align: center"> Price: ${roomshop.items[i].price} ${currencyName}${Chat.plural(roomshop.items[i].price)}</td></tr>`;
+			}
+			display += `</table></div>`;
+			return this.sendReplyBox(display);
+		},
+
+		purchase: "buy",
+		buy: function (target, room, user) {
+			if (!Db.roomshop.has(room.id)) return this.errorReply("Roomshop is not enabled here.");
+			let roomshop = Db.roomshop.get(room.id, {items: {}});
+			if (!roomshop.bank) return this.errorReply(`${room.title} hasn't set a bank yet.`);
+			if (!roomshop.items[toId(target)]) return this.errorReply(`The item "${target}" does not exist.`);
+			if (roomshop.bank === user.userid) return this.errorReply("Bank cannot purchase from the roomshop.");
+			let bank = roomshop.bank;
+			let cost = roomshop.items[toId(target)].price;
+			// Take payments and record payments from the user
+			Economy.readMoney(user.userid, money => {
+				if (money < cost) {
+					this.errorReply(`You do not have enough ${currencyName} to purchase ${target}.`);
+					return;
+				}
+				Economy.writeMoney(user.userid, -cost, () => {
+					Economy.logTransaction(`${user.name} bought ${target} from ${room.title}'s roomshop for ${cost} ${currencyName}${Chat.plural(cost)}.`);
+				});
+				Economy.writeMoney(bank, cost, () => {
+					Economy.logTransaction(`${user.name} bought ${target} from ${room.title}'s roomshop for ${cost} ${currencyName}${Chat.plural(cost)}.`);
+				});
+			});
+			if (FS("logs/roomshops").readdirSync()) FS("logs/roomshops").mkdirpSync();
+			FS(`logs/roomshops/roomshop_${room.id}.txt`).append(`[${new Date().toUTCString()}] ${user.name} has bought ${target} from the roomshop.\n`);
+
+			let msg = `${user.name} has purchased ${target}.`;
+
+			Users.users.forEach(function (user) {
```

Comment:
``for (let u in room.users) {
if (room.auth[u] === "#" || room.auth[u] === "&") {
user.send(`|pm|~${room.title}'s Shop Alert|${user.getIdentity()}|${msg}`);
}
}
No room leaders :p (and no tabbing for me on mobile)

Reply:
Oh true no Room Leaders xD